Generator/Generator/MethodGen.swift

```diff
@@ -62,6 +62,22 @@ func isRefParameterOptional (className: String, method: String, arg: String) ->
     }
 }
 
+/// TODO: The current code generation for passing parameters is both inefficient, and technically unsafe. We don't need
+/// to use nested invocations of withUnsafePointer to generate pointers to multiple arguments, but we can instead generate
+/// a helper version that generates multiple pointer in a single call. For example, if a function that we call using
+/// `gi.object_method_bind_ptrcall()` takes 2 arguments, we can generate the following generic helper:
+/// 
+/// ```
+/// func withUnsafePointers<T1, T2, ReturnType>(
+///     _ p1: UnsafePointer<T1>, _ p2: UnsafePointer<T2>,
+///     _ block: (UnsafePointer<T1>, UnsafePointer<T2>) -> ReturnType
+/// ) -> ReturnType {
+///     block(p1, p2)
+/// }
+/// ```
+/// This reduces the complexity of the generated code, and can be extended to an arbitrary number of parameters.
+/// 
```

Comment:
This is lovely!   I learned something today!

Reply:
I sort of discovered this a few years ago, reasoning that since this is how C APIs are modeled, we could do the same thing in Swift code.